### ISSUE TITLE
Spinning circle instead of authenticator accounts when using Yubikey NEO NFC

### ISSUE THREAD

**Issue by pauby** [Yubico/yubioath-flutter#1374]
When using the Yubikey NEO via NFC in the Android app, I can't see any Authenticator accounts, only a spinning circle. Note that I use the desktop app daily on Linux and it works flawlessly. So the key is not an issue.

I have uploaded a video of it to my [Google Drive](https://drive.google.com/file/d/13fxsbyGU1E2Hf199LlqJm07ijzzidDgX/view?usp=sharing). Hopefully it helps.

Things I've tried:

* Going through the issues here. I found [this one](https://github.com/Yubico/yubioath-flutter/issues/814), but it's closed. There isn't anything in it to 'try'.
* [I tried multiple taps as suggested here](https://github.com/Yubico/yubioath-flutter/issues/967), but it makes no difference.
* I have tried to install and uninstall the app.

I _haven't_ tried the key via USB as I don't have an adapter.

Steps to reproduce:

1. Tap the Yubikey to the phone.
2. Enter the password (I have a PIN on my YubiKey).
3. Get prompted to tap my key again, which I do.
4. Spinning circle. 

After Step 4. I wait for the spinning circle to go away and show the accounts, but that doesn't happen. Instead, the circle eventually stops spinning and leaves the YubiKey Authenticator app as it was when you start it. However, the app no longer works - tapping the NFC key makes the sound, but the app does nothing - so you have to close it and reopen.

Here are the TRAFFIC level logs. I've removed what, I believe, is sensitive.

```
17:01:18:681 [main] DEBUG c.yubico.authenticator.MainActivity -  Setting FLAG_SECURE (disallow screenshots)
17:01:18:706 [main] DEBUG i3.d -  Current app preferences:
17:01:18:718 [main] TRACE i3.a -  Activity alias 'AliasMainActivity' is enabled: 1
17:01:18:719 [main] DEBUG i3.a -  Enabled USB discovery by setting state of AliasMainActivity to ENABLED
17:01:18:719 [main] DEBUG c.yubico.authenticator.MainActivity -  Starting nfc discovery
17:01:18:744 [main] DEBUG i3.d -  registering change listener
17:01:18:778 [main] INFO  com.yubico.authenticator.Log -  Logging initialized, outputting to Android/logcat [android.logger]
17:01:29:468 [main] DEBUG com.yubico.authenticator.Log -  Log level set to TRAFFIC [about]
17:01:37:542 [pool-4-thread-1] DEBUG t3.f -  nfc connection opened
17:01:37:543 [pool-4-thread-1] TRACE t3.f -  sent: <REMOVED>
17:01:37:581 [pool-4-thread-1] TRACE t3.f -  received: <REMOVED>
17:01:37:583 [pool-4-thread-1] DEBUG g4.h -  OATH session initialized (version=1.0.0, isAccessKeySet=true)
17:01:37:591 [pool-4-thread-1] TRACE t3.f -  sent: <REMOVED>
17:01:37:608 [pool-4-thread-1] TRACE t3.f -  received: <REMOVED>
17:01:37:609 [pool-4-thread-1] TRACE t3.f -  sent: <REMOVED>
17:01:37:618 [pool-4-thread-1] TRACE t3.f -  received: <REMOVED> 
17:01:37:619 [pool-4-thread-1] TRACE t3.f -  sent: <REMOVED>
17:01:37:631 [pool-4-thread-1] TRACE t3.f -  received: <REMOVED>
17:01:37:632 [pool-4-thread-1] DEBUG f4.e -  Management session initialized for connection=f, version=3.4.1
17:01:37:633 [pool-4-thread-1] TRACE t3.f -  sent: <REMOVED>
17:01:37:643 [pool-4-thread-1] TRACE t3.f -  received: <REMOVED>
17:01:37:643 [pool-4-thread-1] TRACE t3.f -  sent: <REMOVED>
17:01:37:658 [pool-4-thread-1] TRACE t3.f -  received: <REMOVED>
17:01:37:659 [pool-4-thread-1] DEBUG i4.d -  YubiOTP session initialized for connection=f, version=3.4.1, ledInverted=false
17:01:37:660 [pool-4-thread-1] DEBUG i4.d -  Configuration slot 1: configured=true, touchTriggered=true
17:01:37:661 [pool-4-thread-1] DEBUG i4.d -  Configuration slot 2: configured=true, touchTriggered=false
17:01:37:661 [pool-4-thread-1] TRACE t3.f -  sent: 00 01 10 00 
17:01:37:672 [pool-4-thread-1] TRACE t3.f -  received: 00 4a 60 bb 90 00 
17:01:37:672 [pool-4-thread-1] DEBUG h4.a -  Scan for available ccid applications
17:01:37:673 [pool-4-thread-1] TRACE t3.f -  sent: <REMOVED>
17:01:37:684 [pool-4-thread-1] TRACE t3.f -  received: <REMOVED>
17:01:37:685 [pool-4-thread-1] TRACE t3.f -  sent: <REMOVED>
17:01:37:722 [pool-4-thread-1] TRACE t3.f -  received: <REMOVED> 
17:01:37:723 [pool-4-thread-1] TRACE t3.f -  sent: <REMOVED>
17:01:37:738 [pool-4-thread-1] TRACE t3.f -  received: <REMOVED>
17:01:37:738 [pool-4-thread-1] TRACE t3.f -  sent: <REMOVED>
17:01:37:751 [pool-4-thread-1] TRACE t3.f -  received: <REMOVED>
17:01:37:751 [pool-4-thread-1] TRACE t3.f -  sent: <REMOVED>
17:01:37:761 [pool-4-thread-1] TRACE t3.f -  received: <REMOVED> 
17:01:37:762 [pool-4-thread-1] DEBUG h4.a -  Missing applet AID_U2F_YUBICO, capability U2F
17:01:37:762 [pool-4-thread-1] DEBUG h4.a -  Read info DeviceInfo{config=DeviceConfig{enabledCapabilities={}, autoEjectTimeout=null, challengeResponseTimeout=null, deviceFlags=null}, serialNumber=<REMOVED>, version=3.4.1, formFactor=UNKNOWN, supportedCapabilities={USB=59, NFC=59}, isLocked=false, isFips=false, isSky=false}
17:01:37:772 [main] DEBUG c.y.authenticator.oath.OathManager -  Successfully read Oath session info (and credentials if unlocked) from connected key
17:01:37:783 [pool-4-thread-1] DEBUG t3.f -  nfc connection closed
17:01:48:662 [pool-4-thread-1] DEBUG t3.f -  nfc connection opened
17:01:48:663 [pool-4-thread-1] TRACE t3.f -  sent: <REMOVED>
17:01:48:702 [pool-4-thread-1] TRACE t3.f -  received: <REMOVED> 
17:01:48:703 [pool-4-thread-1] DEBUG g4.h -  OATH session initialized (version=1.0.0, isAccessKeySet=true)
17:01:48:718 [pool-4-thread-1] DEBUG g4.h -  Unlocking session
17:01:48:720 [pool-4-thread-1] TRACE t3.f -  sent: <REMOVED>
17:01:48:759 [pool-4-thread-1] TRACE t3.f -  received: <REMOVED>
17:01:48:760 [pool-4-thread-1] INFO  g4.h -  Calculating all codes for time=1706547718760
17:01:48:761 [pool-4-thread-1] TRACE t3.f -  sent: <REMOVED>
17:01:49:266 [main] DEBUG com.yubico.authenticator.Log -  applet unlocked [android.oath.state]
17:01:49:765 [pool-4-thread-1] ERROR c.y.authenticator.oath.OathManager -  Failed to refresh codesandroid.nfc.TagLostException: Tag was lost.
	at android.nfc.TransceiveResult.getResponseOrThrow(TransceiveResult.java:48)
	at android.nfc.tech.BasicTagTechnology.transceive(BasicTagTechnology.java:154)
	at android.nfc.tech.IsoDep.transceive(IsoDep.java:175)
	at t3.f.A(SourceFile:1)
	at d4.g.l(SourceFile:1)
	at g4.h.n(SourceFile:1)
	at com.yubico.authenticator.oath.OathManager.Q(SourceFile:1)
	at com.yubico.authenticator.oath.OathManager.j(SourceFile:1)
	at com.yubico.authenticator.oath.OathManager$j.a(SourceFile:1)
	at com.yubico.authenticator.oath.OathManager$j.b(SourceFile:1)
	at t3.i.e(SourceFile:1)
	at t3.i.b(SourceFile:1)
	at t3.g.run(SourceFile:1)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:487)
	at java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)
	at java.lang.Thread.run(Thread.java:1012)

17:01:49:766 [main] DEBUG c.y.authenticator.oath.OathManager -  Successfully read Oath session info (and credentials if unlocked) from connected key
17:01:49:768 [pool-4-thread-1] DEBUG t3.f -  nfc connection closed
17:02:19:093 [main] DEBUG i3.d -  unregistered change listener
17:02:19:094 [main] DEBUG c.yubico.authenticator.MainActivity -  Stopped usb discovery
17:02:19:157 [main] DEBUG c.yubico.authenticator.MainActivity -  Stopped nfc discovery
17:02:19:163 [main] TRACE i3.a -  Activity alias 'AliasMainActivity' is enabled: 0
17:02:19:163 [main] DEBUG i3.a -  Disabled USB discovery by setting state of AliasMainActivity to DEFAULT
17:02:19:165 [main] TRACE i3.a -  Activity alias 'AliasNdefActivity' is enabled: 0
17:02:19:165 [main] DEBUG i3.a -  Enabled NFC discovery by setting state of AliasNdefActivity to DEFAULT
17:02:19:181 [main] DEBUG com.yubico.authenticator.Log -  appLifecycleStateChange triggered with: inactive [android.window_state_provider]
17:02:19:182 [main] DEBUG com.yubico.authenticator.Log -  Updated windowState to WindowState(focused: false, visible: false, active: false, hidden: false) [android.window_state_provider]
17:02:19:384 [main] DEBUG com.yubico.authenticator.Log -  appLifecycleStateChange triggered with: hidden [android.window_state_provider]
17:02:19:385 [main] DEBUG com.yubico.authenticator.Log -  Ignoring appLifecycleStateChange [android.window_state_provider]
17:02:19:385 [main] DEBUG com.yubico.authenticator.Log -  appLifecycleStateChange triggered with: paused [android.window_state_provider]
17:02:19:385 [main] DEBUG com.yubico.authenticator.Log -  Ignoring appLifecycleStateChange [android.window_state_provider]
17:03:40:210 [main] TRACE i3.a -  Activity alias 'AliasMainActivity' is enabled: 1
17:03:40:211 [main] DEBUG i3.a -  Enabled USB discovery by setting state of AliasMainActivity to ENABLED
17:03:40:211 [main] DEBUG c.yubico.authenticator.MainActivity -  Starting nfc discovery
17:03:40:275 [main] DEBUG i3.d -  registering change listener
17:03:40:291 [main] DEBUG c.y.authenticator.oath.OathManager -  Removing NFC data after resume.
17:03:40:323 [main] DEBUG com.yubico.authenticator.Log -  appLifecycleStateChange triggered with: hidden [android.window_state_provider]
17:03:40:324 [main] DEBUG com.yubico.authenticator.Log -  Ignoring appLifecycleStateChange [android.window_state_provider]
17:03:40:324 [main] DEBUG com.yubico.authenticator.Log -  appLifecycleStateChange triggered with: inactive [android.window_state_provider]
17:03:40:324 [main] DEBUG com.yubico.authenticator.Log -  Ignoring appLifecycleStateChange [android.window_state_provider]
17:03:40:335 [main] DEBUG com.yubico.authenticator.Log -  Yubikey was detached. [android.devices]
17:03:40:335 [main] DEBUG com.yubico.authenticator.Log -  appLifecycleStateChange triggered with: resumed [android.window_state_provider]
17:03:40:335 [main] DEBUG com.yubico.authenticator.Log -  Updated windowState to WindowState(focused: true, visible: true, active: true, hidden: false) [android.window_state_provider]
17:03:40:335 [main] DEBUG com.yubico.authenticator.Log -  Reading nfc enabled value [android.window_state_provider]
17:03:46:795 [main] INFO  com.yubico.authenticator.Log -  Copying log to clipboard (6.3.1)... [about]
17:03:46:796 [main] DEBUG com.yubico.authenticator.Log -  Getting logs... [android.logger]
```

Other Details:

* Samsung Galaxy S23 Ultra
* Android 14
* Yubico Authenticator Version: 6.3.1

**Comment by AdamVe** [comment]
Hi @pauby, could you please try the latest release 6.4.0 and let us know if that version fixed your issue with YubiKey NEO?

**Comment by pauby** [comment]
So this _has_ and _hasn't_ fixed the issue for me. I feel there is more work to do here.

I have a pin on my Yubikey, so I need to present it twice. When the second 'OAuth' box appears after entering my pin, I put my Yubikey against the phone and when the 'Success'  message appears, I remove it. When I do that, the same behaviour is present: the spinning icon spins forever.

If I do this again, but this time ignoring the 'Success' message and just keep the Yubikey against the phone, I do get the accounts appearing.

There needs to be more messaging around holding the Yubikey until the accounts appear. The 'Success' message is misleading.

**Comment by AdamVe** [comment]
Thank you for the feedback! I understand the issue better now and I could reproduce exactly what you describe - after unlocking, the messages says "Success" before all the accounts have been read - removing the key will leave the app showing just a rotating circle (circular progress bar).

I created a new issue for this: #1443 , please let's continue the discussion there.

**Comment by AdamVe** [comment]
Hi @pauby I would like to highlight a new feature of Yubico Authenticator: in version [7.1.0](https://github.com/Yubico/yubioath-flutter/releases/tag/7.1.0) we added a new overlay which shows that the application is accessing the YubiKey (over NFC) and informs about success/failure - this should solve the issue with misleading "Success" message when the accounts have not been read yet.
